### PR TITLE
Added the validation of adapters functionalities after dlpar test

### PR DESCRIPTION
**Pull request by Naresh-ibm**
added the validation of adapter functionalities after dlpar operation is performed like checking multipath state for FC, ping for n/w adapters, namespace status for nvme etc.

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
nvme adapter
---------------

[root@sys ~]# avocado run --max-parallel-tasks=1 dlpar.py -m dlpar.yaml 
JOB ID     : 4fd39c616a585c6cf890163e0a59ce09c227feb9
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2023-05-30T11.08-4fd39c6/job.log
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-peer_ip-1f2c: STARTED
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-peer_ip-1f2c: PASS (92.33 s)
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-peer_ip-1f2c: STARTED
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-peer_ip-1f2c: PASS (152.93 s)
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-peer_ip-1f2c: STARTED
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-peer_ip-1f2c: PASS (76.17 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2023-05-30T11.08-4fd39c6/results.html
JOB TIME   : 338.13 s
[root@sys ~]#

**Comment by Naresh-ibm** [comment]
for FC
---------

[root@sys ~]# avocado run --max-parallel-tasks=1 dlpar.py -m dlpar.yaml 
JOB ID     : 859ae7678ef5bfc9128b8d23d00e1820b90537a8
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2023-05-30T21.48-859ae76/job.log
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-peer_ip-a17c: STARTED
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-peer_ip-a17c: PASS (142.50 s)
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-peer_ip-a17c: STARTED
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-peer_ip-a17c: CANCEL: Failed to get the location code for the pci device (6.63 s)
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-peer_ip-a17c: STARTED
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-peer_ip-a17c: CANCEL: Failed to get the location code for the pci device (6.62 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2023-05-30T21.48-859ae76/results.html
JOB TIME   : 173.26 s
[root@sys~]#


The failures above are expected as pci_addr gets changed for qlogic adapter which is known one

**Comment by Naresh-ibm** [comment]
[root@sys ~]# avocado run --max-parallel-tasks=1 dlpar.py -m dlpar.yaml
JOB ID     : 69d50a841d88d27fc2081d2bb93932edd95aa7a7
JOB LOG    : /home/naveed/avocado-fvt-wrapper/results/job-2023-05-31T10.24-69d50a8/job.log
Task exception was never retrieved
future: <Task finished coro=<StatusServer.serve_forever() done, defined at /usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py:40> exception=AttributeError("'Server' object has no attribute 'serve_forever'",)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py", line 44, in serve_forever
    await self._server_task.serve_forever()
AttributeError: 'Server' object has no attribute 'serve_forever'
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-url-9bc3: STARTED
 (1/3) dlpar.py:DlparPci.test_dlpar;run-lpar_2-url-9bc3: PASS (149.88 s)
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-url-9bc3: STARTED
 (2/3) dlpar.py:DlparPci.test_drmgr_pci;run-lpar_2-url-9bc3: PASS (157.18 s)
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-url-9bc3: STARTED
 (3/3) dlpar.py:DlparPci.test_drmgr_phb;run-lpar_2-url-9bc3: PASS (139.95 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/naveed/avocado-fvt-wrapper/results/job-2023-05-31T10.24-69d50a8/results.html
JOB TIME   : 452.20 s
[root@sys~]#

File: io/pci/dlpar.py [hunk 1]
```diff
@@ -367,6 +377,101 @@ def changehwres_sriov(self, server, operation, lpar_id, adapter_id,
             self.log.debug(cmd.stderr)
             self.fail("dlpar %s operation failed" % msg)
 
+    def validation_in_os(self):
```
Comment (abdhaleegit) [inline io/pci/dlpar.py:380]: this is not being called anywhere ?

File: io/pci/dlpar.py [hunk 2]
```diff
@@ -367,6 +377,101 @@ def changehwres_sriov(self, server, operation, lpar_id, adapter_id,
...
+                return False
+            return True
+
+        def net_recovery_check():
+            """
+            Checks if the network adapter fuctionality like ping/link_state,
+            after adapter added back.
+            Returns True on propper Recovery, False if not.
+            """
+            self.log.info("entering the net recovery check")
```
Comment (abdhaleegit) [inline io/pci/dlpar.py:426]: this log info is not needed

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
for some adapter it fails to get the location code, can you change the message to saying  slot dlpar unsupported in adapter

File: io/pci/dlpar.py [hunk 3]
```diff
@@ -367,6 +381,101 @@ def changehwres_sriov(self, server, operation, lpar_id, adapter_id,
...
+            local = LocalHost()
+            iface = pci.get_interfaces_in_pci_address(self.pci_device, 'net')
+            networkinterface = NetworkInterface(iface[0], local)
+            if wait.wait_for(networkinterface.is_link_up, timeout=120):
+                if networkinterface.ping_check(self.peer_ip, count=5) is None:
```
Comment (abdhaleegit) [inline io/pci/dlpar.py:435]: peer_ip should be checked before it is use, cancel if not given

**Review by abdhaleegit — APPROVED** [review]
LGTM

File: io/pci/dlpar.py [hunk 4]
```diff
@@ -367,6 +381,101 @@ def changehwres_sriov(self, server, operation, lpar_id, adapter_id,
...
+                              (self.ns_list - current_namespaces))
+                return False
+
+            if err_ns:
+                self.log.info(f"following namespaces not recovered ={err_ns}")
+                return False
+            return True
+
+        if wait.wait_for(is_added, timeout=30):
+            time.sleep(45)
```
Comment (abdhaleegit) [inline io/pci/dlpar.py:463]: one last comment if wait is given than why sleep ?

Comment (Naresh-ibm) [reply io/pci/dlpar.py:463]: this is for splitter limitation. so added extra sleep.
we are expecting enhancement for splitter. so we can remove it later.

**Pull request merged**